clients/src/test/java/org/apache/kafka/clients/consumer/KafkaConsumerTest.java

```diff
@@ -2965,7 +2976,21 @@ public void testSubscriptionOnInvalidTopic(GroupProtocol groupProtocol) {
         KafkaConsumer<String, String> consumer = newConsumer(groupProtocol, time, client, subscription, metadata, assignor, true, groupInstanceId);
         consumer.subscribe(singleton(invalidTopicName), getConsumerRebalanceListener(consumer));
 
-        assertThrows(InvalidTopicException.class, () -> consumer.poll(Duration.ZERO));
+        if (isAyncConsumer(groupProtocol)) {
+            // New consumer poll(ZERO) needs to wait for the event added by a call to poll, to be processed
+            // by the background thread, so it can realize there is invalid topics and then
+            // throw the InvalidTopicException
+            TestUtils.waitForCondition(() -> {
```

Comment:
`isAyncConsumer` is simple enough, and the method needing helper is `TestUtils.waitForCondition` :)

Reply:
Agreed, what about a kind of `assertPollEventuallyThrows`(expectedException, errorMsg) (or better named). We would use it for `NoOffsetForPartitionException`, `InvalidTopicException` and `AuthenticationException`

Reply:
Sorry for misunderstanding. Rewrite it.